### ISSUE TITLE
incompatibility between numpy and shap 

### ISSUE THREAD

**Issue by bpfrd** [shap/shap#3716]
Hi,
I get the below error when trying to import shap. Here is the full traceback

```
!pip install --upgrade numpy shap
# numpy version => 2.0.0
```
```
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
Cell In[146], line 1
----> 1 import shap
      2 shap.initjs()
      3 get_ipython().run_line_magic('matplotlib', 'inline')

File /opt/conda/lib/python3.11/site-packages/shap/__init__.py:45
     43     have_matplotlib = False
     44 if have_matplotlib:
---> 45     from . import plots
     46     from .plots._bar import bar_legacy as bar_plot
     47     from .plots._beeswarm import summary_legacy as summary_plot

File /opt/conda/lib/python3.11/site-packages/shap/plots/__init__.py:6
      3 except ImportError:
      4     raise ImportError("matplotlib is not installed so plotting is not available! Run `pip install matplotlib` to fix this.")
----> 6 from ._bar import bar
      7 from ._beeswarm import beeswarm
      8 from ._benchmark import benchmark

File /opt/conda/lib/python3.11/site-packages/shap/plots/_bar.py:9
      7 from ..utils import format_value, ordinal_str
      8 from ..utils._exceptions import DimensionError
----> 9 from . import colors
     10 from ._labels import labels
     11 from ._utils import (
     12     convert_ordering,
     13     dendrogram_coords,
   (...)
     16     sort_inds,
     17 )

File /opt/conda/lib/python3.11/site-packages/shap/plots/colors/__init__.py:1
----> 1 from ._colors import (
      2     blue_rgb,
      3     gray_rgb,
      4     light_blue_rgb,
      5     light_red_rgb,
      6     red_blue,
      7     red_blue_circle,
      8     red_blue_no_bounds,
      9     red_blue_transparent,
     10     red_rgb,
     11     red_transparent_blue,
     12     red_white_blue,
     13     transparent_blue,
     14     transparent_red,
     15 )
     17 __all__ = [
     18     "blue_rgb",
     19     "gray_rgb",
   (...)
     30     "transparent_red",
     31 ]

File /opt/conda/lib/python3.11/site-packages/shap/plots/colors/_colors.py:23
     21 red_lch = [54., 90., 0.35470565 + 2* np.pi]
     22 gray_lch = [55., 0., 0.]
---> 23 blue_rgb = lch2rgb(blue_lch)
     24 red_rgb = lch2rgb(red_lch)
     25 gray_rgb = lch2rgb(gray_lch)

File /opt/conda/lib/python3.11/site-packages/shap/plots/colors/_colors.py:13, in lch2rgb(x)
     12 def lch2rgb(x):
---> 13     return lab2rgb(lch2lab([[x]]))[0][0]

File /opt/conda/lib/python3.11/site-packages/shap/plots/colors/_colorconv.py:372, in lch2lab(lch)
    346 def lch2lab(lch):
    347     """CIE-LCH to CIE-LAB color space conversion.
    348     LCH is the cylindrical representation of the LAB (Cartesian) colorspace
    349     Parameters
   (...)
    370     >>> img_lab2 = lch2lab(img_lch)
    371     """
--> 372     lch = _prepare_lab_array(lch)
    374     c, h = lch[..., 1], lch[..., 2]
    375     lch[..., 1], lch[..., 2] = c * np.cos(h), c * np.sin(h)

File /opt/conda/lib/python3.11/site-packages/shap/plots/colors/_colorconv.py:387, in _prepare_lab_array(arr)
    385 if shape[-1] < 3:
    386     raise ValueError('Input array has less than 3 color channels')
--> 387 return img_as_float(arr, force_copy=True)

File /opt/conda/lib/python3.11/site-packages/shap/plots/colors/_colorconv.py:993, in img_as_float(image, force_copy)
    972 def img_as_float(image, force_copy=False):
    973     """Convert an image to floating point format.
    974     This function is similar to `img_as_float64`, but will not convert
    975     lower-precision floating point arrays to `float64`.
   (...)
    991     and can be outside the ranges [0.0, 1.0] or [-1.0, 1.0].
    992     """
--> 993     return convert(image, np.floating, force_copy)

File /opt/conda/lib/python3.11/site-packages/shap/plots/colors/_colorconv.py:819, in convert(image, dtype, force_copy, uniform)
    808 itemsize_out = dtypeobj_out.itemsize
    810 # Below, we do an `issubdtype` check.  Its purpose is to find out
    811 # whether we can get away without doing any image conversion.  This happens
    812 # when:
   (...)
    816 #   is a subclass of that type (e.g. `np.floating` will allow
    817 #   `float32` and `float64` arrays through)
--> 819 if np.issubdtype(dtype_in, np.obj2sctype(dtype)):
    820     if force_copy:
    821         image = image.copy()

File /opt/conda/lib/python3.11/site-packages/numpy/__init__.py:397, in __getattr__(attr)
    394     raise AttributeError(__former_attrs__[attr])
    396 if attr in __expired_attributes__:
--> 397     raise AttributeError(
    398         f"`np.{attr}` was removed in the NumPy 2.0 release. "
    399         f"{__expired_attributes__[attr]}"
    400     )
    402 if attr == "chararray":
    403     warnings.warn(
    404         "`np.chararray` is deprecated and will be removed from "
    405         "the main namespace in the future. Use an array with a string "
    406         "or bytes dtype instead.", DeprecationWarning, stacklevel=2)

AttributeError: `np.obj2sctype` was removed in the NumPy 2.0 release. Use `np.dtype(obj).type` instead.
```

best

**Comment by yonicohbgu** [comment]
same here

**Comment by bpfrd** [comment]
Quick temporary fix:
```
np.obj2sctype =  lambda obj: np.dtype(obj).type
import shap
shap.__version__
'0.45.1'
```

**Comment by connortann** [comment]
This issue has been fixed on master by #3712 and #3704 , and will be available in the next release (`0.46`) which will probably be made within the next few days.

**Comment by connortann** [comment]
Version `0.46.0` was released on PyPI today and will be available on conda-forge in due course. Please get in touch if you have any issues.
